lib/vector-buffers/src/disk_v2/record.rs

```diff
@@ -99,27 +111,36 @@ where
 
 impl<'a> Record<'a> {
     /// Creates a [`Record`] from the ID and payload, and calculates the checksum.
-    pub fn with_checksum(id: u64, payload: &'a [u8], checksummer: &Hasher) -> Self {
-        let checksum = generate_checksum(checksummer, id, payload);
+    pub fn with_checksum(id: u64, metadata: u32, payload: &'a [u8], checksummer: &Hasher) -> Self {
+        let checksum = generate_checksum(checksummer, id, metadata, payload);
```

Comment:
Will updating the checksum here have any impact to existing v2 disk buffers?

Reply:
Yes, but so does adding the metadata field itself.  This is definitely a breaking change, but we're intentionally doing it before even announcing the disk buffer v2 stuff as beta.